web3/_utils/abi.py

```diff
@@ -302,6 +313,85 @@ def merge_args_and_kwargs(function_abi, args, kwargs):
         return tuple()
 
 
+TUPLE_TYPE_STR_RE = re.compile(r'^(tuple)(\[([1-9][0-9]*)?\])?$')
+
+
+def get_tuple_type_str_parts(s: str) -> Optional[Tuple[str, Optional[str]]]:
+    """
+    Takes a JSON ABI type string.  For tuple type strings, returns the separated
+    prefix and array dimension parts.  For all other strings, returns ``None``.
+    """
+    match = TUPLE_TYPE_STR_RE.match(s)
+
+    if match is not None:
+        tuple_prefix = match.group(1)
+        tuple_dims = match.group(2)
+
+        return tuple_prefix, tuple_dims
+
+    return None
+
+
+def _convert_abi_input(comp, arg):
+    """
+    Converts an argument ``arg`` corresponding to an ABI component ``comp``
```

Comment:
`component` here seems confusing because each element of a tuple is called `component`. Maybe this is a `arg_type`, or `arg_spec` since it includes a name?